src/raglite/_database.py

```diff
@@ -331,53 +344,69 @@ def create_database_engine(config: RAGLiteConfig | None = None) -> Engine:
         with Session(engine) as session:
             metrics = {"cosine": "cosine", "dot": "ip", "euclidean": "l2", "l1": "l1", "l2": "l2"}
             session.execute(
-                text("""
-                CREATE INDEX IF NOT EXISTS keyword_search_chunk_index ON chunk USING GIN (to_tsvector('simple', body));
-                """)
+                text(
+                    """
+                    CREATE INDEX IF NOT EXISTS keyword_search_chunk_index ON chunk USING GIN (to_tsvector('simple', body));
+                    """
+                )
             )
-            session.execute(
-                text(f"""
+            base_sql = f"""
                 CREATE INDEX IF NOT EXISTS vector_search_chunk_index ON chunk_embedding
                 USING hnsw (
                     (embedding::halfvec({embedding_dim}))
                     halfvec_{metrics[config.vector_search_index_metric]}_ops
                 );
                 SET hnsw.ef_search = {20 * 4 * 8};
-                SET hnsw.iterative_scan = {'relaxed_order' if config.reranker else 'strict_order'};
-                """)
-            )
+            """
+            # Add iterative scan if version >= 0.8.0
+            pgvector_version = _get_pgvector_version(session)
+            if pgvector_version and version.parse(pgvector_version) >= version.parse("0.8.0"):
+                sql = f"""{base_sql};
+                    SET hnsw.iterative_scan = {'relaxed_order' if config.reranker else 'strict_order'};
+                    """
+            else:
+                sql = f"{base_sql};"
+            session.execute(text(sql))
```

Comment:
Comments:
- No need to add `;` to the end of base_sql (it already ends with a `;`).
- I could not modify it fully in the suggestion below, but I would rename `base_sql` to `create_vector_index_sql`.

```suggestion
            # Enable iterative scan from pgvector version >= 0.8.0.
            if _pgvector_version(session) >= version.parse("0.8.0"):
                create_vector_index_sql += f"\nSET hnsw.iterative_scan = {'relaxed_order' if config.reranker else 'strict_order'};"
            session.execute(text(create_vector_index_sql))
```

Reply:
Good catch, I saw it after committing. Not a big deal as SQL tolerates several ';;' but I will change it.